src/llm/legacy/legacy_executor.hpp

```diff
@@ -0,0 +1,136 @@
+#include <thread>
+#include <future>
+
+namespace ovms { 
+struct LegacyServableExecutionContext : public GenAiServableExecutionContext {
+    ov::genai::EncodedResults results;
+    std::promise<void> readySignal;
+    std::future<void> finished = readySignal.get_future();
+};
+
+struct LegacyExecutor {
+    // For logging purposes we could have more information about graph and node here
+    std::condition_variable cv;
+    std::queue<std::shared_ptr<LegacyServableExecutionContext>> requests;
+    std::mutex queueMutex;
+    std::shared_ptr<ov::genai::LLMPipeline> pipe;
+
+    LegacyExecutor(std::shared_ptr<ov::genai::LLMPipeline> pipe) {
+        this->pipe = std::move(pipe);
+    }
+
+    bool hasRequests() {
+        return (requests.size() > 0);
+    }
+
+    void step() {
+        OVMS_PROFILE_FUNCTION();
+        requests.front()->results = pipe->generate(requests.front()->inputIds, requests.front()->apiHandler->createGenerationConfig(), requests.front()->textStreamer);
+        std::unique_lock<std::mutex> lock(queueMutex);
+        requests.front()->readySignal.set_value();
+        requests.pop();
+    }
+
+    void waitForRequests(std::atomic<bool>* receivedEndSignal) {
+        std::unique_lock<std::mutex> lock(queueMutex);
+        cv.wait(lock, [this, receivedEndSignal] { return (requests.size() > 0 || *receivedEndSignal); });
+    }
+
+    void addRequest(std::shared_ptr<LegacyServableExecutionContext> request) {
+        std::lock_guard<std::mutex> guard(queueMutex);
+        requests.push(request);
+        cv.notify_one();
+    }
+
+    void notify() {
+        std::unique_lock<std::mutex> lock(queueMutex);
+        cv.notify_one();
+    }
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-but-set-variable"
+    // void printMetrics() {
+    //     ov::genai::PipelineMetrics metrics = pipe->get_metrics();
+    //     SPDLOG_LOGGER_INFO(llm_executor_logger, "All requests: {}; Scheduled requests: {}; Cache usage {:.1f}%;",
+    //         metrics.requests, metrics.scheduled_requests, metrics.cache_usage);
+    // }
+};
+#pragma GCC diagnostic pop
+
+class LegacyExecutorWrapper {
+    LegacyExecutor legacyExecutor;
+    std::thread legacyExecutorThread;
+    std::atomic<bool> finishExecutorThread = false;
+
+    static void run(LegacyExecutor* legacyExecutor, std::atomic<bool>* receivedEndSignal) {
+        const uint8_t printMetricsEveryNumberOfSteps = 10;
+        uint8_t stepCounter = 0;
+        while (!(*receivedEndSignal)) {
+            try {
+                if (stepCounter == printMetricsEveryNumberOfSteps) {
+                    // legacyExecutor->printMetrics();
+                    stepCounter = 0;
+                }
+                if (legacyExecutor->hasRequests()) {
+                    stepCounter++;
+                    legacyExecutor->step();
+                } else {
+                    SPDLOG_LOGGER_INFO(llm_executor_logger, "All requests: {}; Scheduled requests: {};", 0, 0);
```

Comment:
Scheduled requests will always be 1 in non CB flow. Let's print "Awaiting requests: <size_of_the_requests_queue>" instead.